manifests/install.pp

```diff
@@ -1,229 +1,72 @@
-  String $dashboard_cert_file             =  $puppet_metrics_dashboard::params::dashboard_cert_file,
-  String $dashboard_cert_key              =  $puppet_metrics_dashboard::params::dashboard_cert_key,
-  Boolean $overwrite_dashboards           =  $puppet_metrics_dashboard::params::overwrite_dashboards,
-  String $overwrite_dashboards_file       =  $puppet_metrics_dashboard::params::overwrite_dashboards_file,
-  String $influx_db_service_name          =  $puppet_metrics_dashboard::params::influx_db_service_name,
-  Array[String] $influxdb_database_name   =  $puppet_metrics_dashboard::params::influxdb_database_name,
-  String $grafana_version                 =  $puppet_metrics_dashboard::params::grafana_version,
-  Integer $grafana_http_port              =  $puppet_metrics_dashboard::params::grafana_http_port,
-  String $influx_db_password              =  $puppet_metrics_dashboard::params::influx_db_password,
-  String $grafana_password                =  $puppet_metrics_dashboard::params::grafana_password,
-  Boolean $enable_kapacitor               =  $puppet_metrics_dashboard::params::enable_kapacitor,
-  Boolean $enable_chronograf              =  $puppet_metrics_dashboard::params::enable_chronograf,
-  Boolean $enable_telegraf                =  $puppet_metrics_dashboard::params::enable_telegraf,
-  Boolean $configure_telegraf             =  $puppet_metrics_dashboard::params::configure_telegraf,
-  Boolean $consume_graphite               =  $puppet_metrics_dashboard::params::consume_graphite,
-  Array[String] $master_list              =  $puppet_metrics_dashboard::params::master_list,
-  Array[String] $puppetdb_list            =  $puppet_metrics_dashboard::params::puppetdb_list,
-  String $influxdb_urls                   =  $puppet_metrics_dashboard::params::influxdb_urls,
-  String $telegraf_db_name                =  $puppet_metrics_dashboard::params::telegraf_db_name,
-  Integer[1] $telegraf_agent_interval     =  $puppet_metrics_dashboard::params::telegraf_agent_interval,
-  Integer[1] $http_response_timeout       =  $puppet_metrics_dashboard::params::http_response_timeout,
-) inherits puppet_metrics_dashboard::params {
-
-  # Enable Telegraf if `configure_telegraf` is true.
-  $_enable_telegraf = $configure_telegraf ? {
-    true    => true,
-    default => $enable_telegraf
-  }
-
-  class { 'puppet_metrics_dashboard::repos':
-    manage_repos => $manage_repos,
-  }
-
+class puppet_metrics_dashboard::install {
   package { 'influxdb':
     ensure  => present,
     require => Class['puppet_metrics_dashboard::repos'],
   }
 
-  if $consume_graphite {
-
-    file {'/etc/influxdb/influxdb.conf':
-      ensure  => file,
-      owner   => 0,
-      group   => 0,
-      content => epp('puppet_metrics_dashboard/influxdb.conf.epp',
-        {master_list => $master_list}),
-      notify  => Service[$influx_db_service_name],
-      require => Package['influxdb'],
+  if $puppet_metrics_dashboard::use_dashboard_ssl {
```

Comment:
I get a dependency cycle trying to set `use_dashboard_ssl => true`

```
Error: Found 1 dependency cycle:
(File[/etc/grafana/l12g77n1xyii1zv.delivery.puppetlabs.net_cert.pem] => Class[Puppet_metrics_dashboard::Install] => Class[Puppet_metrics_dashboard::Config] => Class[Puppet_metrics_dashboard::Config] => Class[Puppet_metrics_dashboard::Service] => Service[influxd] => Class[Grafana] => Class[Grafana::Install] => Package[grafana] => File[/etc/grafana/l12g77n1xyii1zv.delivery.puppetlabs.net_cert.pem])
```

Reply:
Didn't think about testing that one... let me take a look.

Reply:
It seems the acceptance test is a bit too ambitious for this PR... it would require implementing the use of certs other than those created by the puppet agent . I got it sorta working locally but testing is another story. Maybe you can give it another go @suckatrash in pooler like you were before. The unit test makes me think I fixed the problem.